virttest/libvirt_xml/nodedev_xml.py

```diff
@@ -152,16 +152,66 @@ class MdevXML(CAPXML):
     """
     class for capability whose type is mdev
     """
-    __slots__ = ('type_id', 'uuid')
+    __slots__ = ('type_id', 'uuid', 'attrs')
 
     def __init__(self, virsh_instance=base.virsh):
         accessors.XMLAttribute('type_id', self, parent_xpath='/',
                                tag_name='type', attribute='id')
         accessors.XMLElementText('uuid', self, parent_xpath='/',
                                  tag_name='uuid')
+        accessors.XMLElementList(property_name='attrs',
+                                 libvirtxml=self,
+                                 parent_xpath='/',
+                                 marshal_from=self.marshal_from_attrs,
+                                 marshal_to=self.marshal_to_attrs,
+                                 has_subclass=True)
         super(MdevXML, self).__init__(virsh_instance=virsh_instance)
         self.xml = (' <capability type=\'mdev\'></capability>')
 
+    @staticmethod
+    def marshal_from_attrs(item, index, libvirtxml):
+        """
+        Convert an AttrXML object to attr tag and xml element.
+        """
+        if isinstance(item, AttrXML):
+            return 'attr', item
+        elif isinstance(item, dict):
+            attr = AttrXML()
+            attr.setup_attrs(**item)
+        else:
+            raise xcepts.LibvirtXMLError("Expected a list of capability attr "
+                                         " not a %s" % item)
+
+    @staticmethod
+    def marshal_to_attrs(tag, new_treefile, index, libvirtxml):
+        """
+        Convert an attr tag xml element to an object of AttrXML.
+        """
+        if tag != 'attr':
+            return None    # Don't convert this item
+        newone = AttrXML(virsh_instance=libvirtxml.virsh)
+        newone.xmltreefile = new_treefile
+        return newone
+
+
+class AttrXML(base.LibvirtXMLBase):
```

Comment:
According to the xml of <attr>, seems no need to define this class, we could simply define the accessor with 'has_subclass = False' and with shorter marshal functions, if <attr> doesn't have sub-element. What do you think?

Reply:
Like this:
```
accessors.XMLElementList('vcpupins', self, parent_xpath='/',
                                 marshal_from=self.marshal_from_vcpupins,
                                 marshal_to=self.marshal_to_vcpupins)
```

Reply:
Thank you for the suggestion. It works well.

Reply:
Thanks for taking my suggestion, btw there's inspekt issue needs to be resovle.